### PR TITLE
MR-688 - Bulk close work order confirmation modal

### PR DESCRIPTION
**Pull request by ange47rm**
Implemented confirmation modal when a user submits a request to close work orders

image

### CODE REVIEW

File: src/components/BackOffice/CloseWorkOrders/index.js [hunk 1]
```diff
@@ -88,7 +88,39 @@ const CloseWorkOrders = () => {
     return newErrors
   }
 
-  const handleSubmit = (event) => {
+  const renderConfirmationModal = () => {
```
Comment (Asura5) [inline src/components/BackOffice/CloseWorkOrders/index.js:91]: Remember there are instances where 300+ work orders are being closed at the same time

Comment (ange47rm) [reply src/components/BackOffice/CloseWorkOrders/index.js:91]: Thanks, might be worth removing the list of work orders!

File: src/components/BackOffice/CloseWorkOrders/index.js [hunk 2]
```diff
@@ -132,36 +173,39 @@ const CloseWorkOrders = () => {
...
                   onClick={() => setFormSuccess(null)}
                 >
-                  Bulk-close more workOrders
+                  Bulk-close more Work Orders
                 </a>
               </p>
             </div>
           ) : (
-            <form onSubmit={handleSubmit}>
+            <form onSubmit={validateForm}>
```
Comment (LBHCallumM) [inline src/components/BackOffice/CloseWorkOrders/index.js:201]: I don't think validateForm entirely explains what this method does. HandleSubmit is much better

File: src/components/BackOffice/CloseWorkOrders/index.js [hunk 3]
```diff
@@ -88,7 +88,44 @@ const CloseWorkOrders = () => {
...
+        <Dialog
+          title="Are you sure?"
+          isOpen={showDialog}
+          onDismiss={() => setShowDialog(false)}
+          data-test="confirmation-modal"
+        >
+          <h3 className="lbh-heading-h3">
+            Are you sure you want to change the status of these Work Orders to "
+            {selectedOption == 'CloseToBase' ? 'Completed' : 'Cancelled'}"?
+          </h3>
```
Comment (LBHCallumM) [inline src/components/BackOffice/CloseWorkOrders/index.js:103]: The wording needs to be more direct. I don't think this is enough to scare people into being 100% sure they're doing the right thing.

I suggest something more like "Permanently Cancel WorkOrders", or "Complete WorkOrders". Also, the confirmation button should say something like 'Delete workOrders'. Confirm is too vague.

This article is pretty helpful:
https://uxdesign.cc/are-you-sure-you-want-to-do-this-microcopy-for-confirmation-dialogues-1d94a0f73ac6

Comment (LBHCallumM) [reply src/components/BackOffice/CloseWorkOrders/index.js:103]: Also, could you add this for the two other forms

File: src/components/BackOffice/CloseWorkOrders/index.js (hunk 2 above)
Comment (ange47rm) [reply src/components/BackOffice/CloseWorkOrders/index.js:201]: I needed to link the "Yes button" of the modal to an action, so I've put validateForm first, to validate the form for errors, and if there are no errors, then the modal is shown. At this point you know the form has been populated correctly and there are no errors. If then the user presses "Yes/Confirm" the network request is fired. I think it's ok as is but if you've got any suggestions on how to name this method let me know your suggestions

File: src/components/BackOffice/CloseWorkOrders/index.js [hunk 4]
```diff
@@ -88,7 +88,24 @@ const CloseWorkOrders = () => {
...
 
-  const handleSubmit = (event) => {
+  const renderConfirmationModal = () => {
+    if (showDialog) {
+      return (
+        <ConfirmationModal
+          title={'Permanently close Work Orders?'}
+          showDialog
+          setShowDialog={setShowDialog}
+          modalText={`The status of the selected Work Orders will change to "${
```
Comment (Asura5) [inline src/components/BackOffice/CloseWorkOrders/index.js:98]: Work Orders is capitalised throughout - is there a reason for this? It's not a proper noun.

Comment (ange47rm) [reply src/components/BackOffice/CloseWorkOrders/index.js:98]: I agree it shouldn't be, but I've capitalised it to keep it consistent with the rest of the solution. Happy to change it everywhere to "work orders"

Comment (Asura5) [reply src/components/BackOffice/CloseWorkOrders/index.js:98]: yeah change it to no caps

Comment (ange47rm) [reply src/components/BackOffice/CloseWorkOrders/index.js:98]: Will do

Comment (ange47rm) [reply src/components/BackOffice/CloseWorkOrders/index.js]: Done

**Review by Asura5 — APPROVED** [review]

**Pull request merged**